### PR TITLE
cufflinks 2.2.1: Modernize. Add doi, tag

### LINKED ISSUE #1982: cufflinks failed to build on OS X 10.10, Xcode 6.2
Hello, I tried to build cufflinks on a brand new brew install on my MacBook Pro with OS X 10.10. Xcode and CLT are updated to the newest version 6.2. 
Dependencies homebrew/versions/boost149, samtools-0.1, eigen installed fine. 
Installing cufflinks reported:
==> Installing cufflinks
==> Downloading http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/cufflinks-2.2.1.tar.gz
######################################################################## 100.0%
==> ./configure --prefix=/usr/local/Cellar/cufflinks/2.2.1 --mandir=/usr/local/Cellar/cufflinks/2.2.1/share/man
==> make
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [gtf_to_sam] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

brew updated was run before
brew doctor reported
Your system is ready to brew.

A Gist is available here:
https://gist.github.com/1639834c83c2792bc17e

Thank you!

### PR DESCRIPTION
**Pull request by saketkc**
Probably fixes(bottles)  #1982 #1983

### CODE REVIEW

File: cufflinks.rb [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-  depends_on "homebrew/versions/boost149"    => :build
+  depends_on "homebrew/versions/boost149" => :build
   depends_on "samtools-0.1" => :build
-  depends_on "eigen"    => :build
+  depends_on "eigen" => :build
+
+  fails_with :clang do
+    build 600
+    cause "Clang not supported"
+  end
```
Comment (sjackman) [inline cufflinks.rb]: Did cufflinks previously build with clang, and is this failure a new development?

**Comment by saketkc** [comment]
Unfortunately I have never been able to build cufflinks. 
See https://github.com/Homebrew/homebrew-science/issues/1240

**Comment by iml** [comment]
FWIW, I managed to build it using Ubuntu's libboost-all-dev (1.55). 

```diff
@@ -5,13 +5,13 @@ class Cufflinks < Formula
   url "http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/cufflinks-2.2.1.tar.gz"
   sha1 "2b1b3a8f12cd2821ffc74ffbdd55cb329f37cbbb"
 
-  depends_on "homebrew/versions/boost149"    => :build
+  # depends_on "homebrew/versions/boost149"    => :build
   depends_on "samtools-0.1" => :build
   depends_on "eigen"    => :build
 
   def install
     ENV['EIGEN_CPPFLAGS'] = "-I#{Formula["eigen"].opt_include}/eigen3"
-    ENV.append 'LIBS', '-lboost_system-mt -lboost_thread-mt'
+    ENV.append 'LIBS', '-lboost_system -lboost_serialization -lboost_thread'
     cd 'src' do
       # Fixes 120 files redefining `foreach` that break building with boost
       # See http://seqanswers.com/forums/showthread.php?t=16637

**Comment by iml** [comment]
Got it to build on Mac bottles. Linux will still need some finesse, as `homebrew/versions/boost155` doesn't build there and I think the -mt suffixes don't apply.

**Comment by saketkc** [comment]
I just pushed an update for both [`boost155`](https://github.com/saketkc/linuxbrew-versions/commit/337a30eda6c72151e0336516bfb387f199714870) and [`boost149`](https://github.com/saketkc/linuxbrew-versions/commit/4305e090fe8f99fb59fb94ba3a73c51b08b67f86) on `linuxbrew-versions`, a fork of `homebrew-versions`.

**Comment by saketkc** [comment]
Thanks @iml, it finally builds on linux too. For the record, I monkey patched `boost155`: https://github.com/saketkc/linuxbrew-versions/blob/337a30eda6c72151e0336516bfb387f199714870/boost155.rb
```
$ brew install cufflinks.rb 
==> Installing cufflinks from homebrew/homebrew-science
==> Downloading http://cole-trapnell-lab.github.io/cufflinks/assets/downloads/cufflinks-2.2.1.tar.gz
Already downloaded: /home/saket/.cache/Homebrew/cufflinks-2.2.1.tar.gz
==> ./configure --prefix=/home/saket/.linuxbrew/Cellar/cufflinks/2.2.1 --mandir=/home/saket/.linuxbrew/Cellar/cufflinks/2.2.1/share/man
==> make
==> make install
/home/saket/.linuxbrew/Cellar/cufflinks/2.2.1: 13 files, 143M, built in 64 seconds
```

**Pull request closed without merging**